torchgen/model.py

```diff
@@ -965,7 +989,10 @@ def __post_init__(self) -> None:
         if self.structured:
             # For now, structured composite kernels are not supported (need some
             # design work to figure out how to make the composite case work)
-            assert not self.out.has_composite_implicit_autograd_kernel
+            assert (
+                not self.out.has_composite_implicit_autograd_kernel
+                and not self.out.has_composite_implicit_autograd_nested_tensor_kernel
```

Comment:
Is it the case that "if an op has a CompositeImplicitAutogradNestedTensor kernel, then it always has a CompositeImplicitAutograd kernel"?

If it is, then we can probably take advantage of that to simplify this: just assert that in `model.py`, and you don't need this extra check.

Reply:
I guess usually it is, but there may be ops designated to nested tensor (e.g. `to_padded_tensor`)

@drisspg @jbschlosser what do you guys think?

Reply:
For all our current ops I think that is the case but I am not 100% that is the case forever

Reply:
we do need this I wouldn't change

Reply:
Agreed, now that we have https://github.com/pytorch/pytorch/pull/83371/files#r950576993